src/backend/routers/conversation.py

```diff
@@ -167,6 +169,80 @@ async def delete_conversation(
     return DeleteConversation()
 
 
+@router.get(":search", response_model=list[ConversationWithoutMessages])
+async def search_conversations(
+    query: str,
+    session: DBSessionDep,
+    request: Request,
+    offset: int = 0,
+    limit: int = 100,
+    agent_id: str = None,
+) -> list[ConversationWithoutMessages]:
+    """
+    Search conversations by title.
+
+    Args:
+        query (str): Query string to search for in conversation titles.
+        session (DBSessionDep): Database session.
+        request (Request): Request object.
+
+    Returns:
+        list[ConversationWithoutMessages]: List of conversations that match the query.
+    """
+    user_id = get_header_user_id(request)
+    deployment_name = request.headers.get("Deployment-Name", "")
+    model_deployment = get_deployment(deployment_name)
+    trace_id = request.state.trace_id if hasattr(request.state, "trace_id") else None
+
+    conversations = conversation_crud.get_conversations(
+        session, offset=offset, limit=limit, user_id=user_id, agent_id=agent_id
+    )
+
+    if not conversations:
+        return []
+
+    rerank_documents = []
+    for conversation in conversations:
+        chatlog = extract_details_from_conversation(conversation)
+
+        document = f"Title: {conversation.title}\n"
+        if len(chatlog.strip()) != 0:
+            document += "\nChatlog:\n{chatlog}"
+
+        rerank_documents.append(document)
+
+    # if rerank is not enabled, filter out conversations that don't contain the query
+    if not model_deployment.rerank_enabled:
+        filtered_conversations = []
+
+        for rerank_document, conversation in zip(rerank_documents, conversations):
+            if query.lower() in rerank_document.lower():
+                filtered_conversations.append(conversation)
+
+        return filtered_conversations
+
+    # Rerank documents
+    res = model_deployment.invoke_rerank(
```

Comment:
We should do some validation on the `res` here since not all model deployments have a proper `invoke_rerank` method. Good example is `BedrockDeployment` which just returns None

Reply:
I think all the model deployments have a flag `rerank_enabled` which I'm using to check that in the condition above, but I'll add some extra validation here just in case

Reply:
@lusmoura ah yep good point, thanks for adding validation